### PR TITLE
build: try creating the go bin directory

### PR DESCRIPTION
**Pull request by robmonte**
Currently if the $GOPATH/bin directory does not exist, the binary will fail to be copied to the location.

I don't believe an if statement is needed to check for directory existence before the mkdir command, it simply does nothing when the directory already exists.

Closes https://github.com/hashicorp/vault/issues/7170

### CODE REVIEW

**Review by raymonstah — APPROVED** [review]
Curious if this stemmed from an open issue.

**Comment by robmonte** [comment]
> Curious if this stemmed from an open issue.

@raymonstah Not that I've seen, no. I encountered this myself when setting up a fresh EC2 instance to build and run Vault, and thought it'd be a good idea. However I came across this old issue: https://github.com/hashicorp/vault/issues/7170

It kinda makes sense to me that some users with GOBIN set may want it in that location. Though I think this location is usually meant for Go packages when using `go install` and not technically for build artifacts? At least that's my understanding if someone can correct me.

Curious what others would think of something like this for line 53:
`IFS=: FIRST=($GOPATH) MAIN_GOPATH=${GOBIN:-$FIRST}`

This sets MAIN_GOPATH to the values in this order:
1. If GOBIN is set, always set MAIN_GOPATH to GOBIN (ignore GOPATH)
2. If _only_ GOPATH is set, set MAIN_GOPATH to the first element in GOPATH
3. If neither are set, MAIN_GOPATH remains unset (uses Go's default of $HOME/go)

File: scripts/build.sh [hunk 1]
```diff
@@ -50,11 +50,13 @@ ${GO_CMD} build \
...
 # Move all the compiled things to the $GOPATH/bin
 OLDIFS=$IFS
-IFS=: MAIN_GOPATH=($GOPATH)
+IFS=: FIRST=($GOPATH) BIN_PATH=${GOBIN:-${FIRST}/bin}
 IFS=$OLDIFS
 
-rm -f ${MAIN_GOPATH}/bin/vault
-cp bin/vault ${MAIN_GOPATH}/bin/
+# Ensure the go bin folder exists
+mkdir -p ${BIN_PATH}
```
Comment (cipherboy) [inline scripts/build.sh:57]: I think I have been hit by this before in some weird environment too, so thank you for fixing this!

Comment (robmonte) [reply scripts/build.sh:57]: Yeah, it's minor issue but a nice fix for users, particularly if they aren't as familiar with Go's paths.

**Review by ncabatoff — APPROVED** [review]
This might warrant a changelog entry, not so much for vault users as for vault packagers; conceivably this might break someone's build script.

**Pull request merged**